### ISSUE TITLE
semantic: failed to produce syntax error for undeclared private field access

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#7582]
This does not produce a syntax error, but it should because there is no `#x` property.

```js
class C {
  x = 1;
  method() {
    obj.#x;
  }
}
```

It seems syntax checker is confusing public property `x` with private property `#x`. If you rename the public property, it works as expected.

[Oxc playground](https://playground.oxc.rs/#eNqNTs1qwzAMfhWjXTYIZS3skrHToE+Ri+MqnYstFclZE0LevXbS9FwQSJ/0/WgCBzW4YFXNr5kaMmYwP2b/XaaI6Y9P7x/r3hhuL7u3YTnNDeVq6KE8rJTxdSlUwFBPID2VpiMlO0CdpMcKgqcEdWeDZqCOr7hddIwthw0lsaQdS3yQ5wquVhRlseReHB59QLIxG0BCTbuLQmaVgIWV56fJKktWzpjTAfXwuf8qbMcnPGP5M4PoyXd+EzumJByOgW9F/I/Ssuaw9Z15vgPd/nWN)

**Comment by camc314** [comment]
question: surely this is ok because you are accessing `#x` on `obj` not on `this`?

TS doesn't report a syntax error in this case:
```
var obj: any;

class C {
  x = 1;
  method() {
    obj.#x;
  }
}
```

instead you get "Property `#x` does not exist on type 'any'"

**Comment by overlookmotel** [comment]
Private properties are lexically scoped. It's always a syntax error to use `#x`, unless that code is inside a class which defines a property `#x`. Try running the example above in NodeJS.

The fact that `obj` may not be an instance of a class which defines a private property `#x` is a separate problem - but that's a *runtime* error, not a syntax error. TypeScript makes it a type error too, but that's a TS-specific thing, and again a separate issue.

I'll admit, none of this is obvious! Everything around private properties is fiendishly complicated. Hence why I've been working on the class properties transform for weeks, and it's still not complete...
